llvm/lib/Target/EVM/EVMAsmPrinter.cpp

```diff
@@ -315,6 +315,22 @@ void EVMAsmPrinter::emitWideRelocatableSymbol(const MachineInstr *MI) {
 }
 
 void EVMAsmPrinter::emitEndOfAsmFile(Module &) {
+  // Construct a local MCSubtargetInfo here.
```

Comment:
I'd moved it to emission and wrote something alike:
// Build a local MCSubtargetInfo as MachineFunction info is unavailable at this global phase.
// It's also not needed to emit INVALID instruction at the end of the .text section.

Reply:
Done.